ts/screens/wallet/payment/ConfirmPaymentMethodScreen.tsx

```diff
@@ -257,119 +275,218 @@ const ConfirmPaymentMethodScreen: React.FC<Props> = (props: Props) => {
 
-  const showFeeContextualHelp = typeof fee !== "undefined" && fee > 0;
+  const formattedSingleAmount = formatNumberCentsToAmount(
+    verifica.importoSingoloVersamento,
+    true
+  );
+  const formattedTotal = formatNumberCentsToAmount(totalAmount, true);
+  const formattedFees = formatNumberCentsToAmount(fee ?? 0, true);
+
+  // Retrieve all the informations needed by the
+  // user interface based on the payment method
+  // selected by the user.
+  const paymentMethodInfo = getPaymentMethodInfo(
+    paymentMethod,
+    props.isPaypalEnabled
+  );
+
+  const paymentMethodSubject = paymentMethodInfo
+    .map(({ subject }) => subject)
+    .getOrElse("");
+
+  const paymentMethodExpiration = paymentMethodInfo
+    .map(({ expiration }) => expiration)
+    .getOrElse("");
+
+  const paymentMethodCaption = paymentMethodInfo
+    .map(({ caption }) => caption)
+    .getOrElse("");
+
+  const paymentMethodLogo = paymentMethodInfo
+    .map(({ logo }) => logo)
+    .getOrElse(<View />);
+
+  const paymentMethodA11yLabel = paymentMethodInfo
+    .map(({ accessibilityLabel }) => accessibilityLabel)
+    .getOrElse("");
+
+  // It should be possible to change PSP only when the user
+  // is not paying using PayPal or the relative flag is
+  // enabled.
+  const canChangePsp = !isPayingWithPaypal || editPaypalPspEnabled;
+
+  // The privacy url needed when paying
+  // using PayPal.
+  const privacyUrl = props.paypalSelectedPsp?.privacyUrl;
+
+  // Retrieve the PSP name checking if the user is
+  // paying using PayPal or another method. The PSP
+  // could always be `undefined`.
+  const pspName = fromNullable(
+    isPayingWithPaypal
+      ? props.paypalSelectedPsp?.ragioneSociale
+      : wallet.psp?.businessName
+  )
+    .map(name => `${I18n.t("wallet.ConfirmPayment.providedBy")} ${name}`)
+    .getOrElse(I18n.t("payment.noPsp"));
 
   return (
     <BaseScreenComponent
       goBack={props.onCancel}
       headerTitle={I18n.t("wallet.ConfirmPayment.header")}
       contextualHelpMarkdown={contextualHelpMarkdown}
       faqCategories={["payment"]}
+      backButtonTestID="cancelPaymentButton"
     >
       <SafeAreaView style={styles.flex}>
-        <Content noPadded={true} bounces={false}>
-          <PaymentBannerComponent
-            currentAmount={verifica.importoSingoloVersamento}
-            paymentReason={paymentReason}
-            fee={fee as ImportoEuroCents}
-          />
-          <View style={styles.padded}>
-            <View spacer={true} />
-            <PaymentMethodCard
-              isPaypalEnabled={props.isPaypalEnabled}
-              paymentMethod={props.getPaymentMethodById(wallet.idWallet)}
+        <Content noPadded={true}>
+          <View style={IOStyles.horizontalContentPadding}>
+            <View spacer />
+
+            <View
+              style={styles.totalContainer}
+              accessibilityRole="header"
+              accessibilityLabel={`Totale ${formattedTotal}`}
```

Comment:
Should it be localized?

Reply:
Definitely 😅 it should be fixed here: https://github.com/pagopa/io-app/pull/3840/commits/31a92324ea4415a9ce3fa91a77d808a0c0c03c32